src/Checkbox.js

```diff
@@ -1,170 +1,147 @@
-import React from "react";
-import { View, TouchableOpacity, StyleSheet, Image } from "react-native";
-import PropTypes from "prop-types";
+/* eslint-disable object-curly-newline */
+import React from 'react';
+import { View, TouchableOpacity, StyleSheet, Image } from 'react-native';
+import PropTypes from 'prop-types';
 // galio dependency
-import { Icon, Text } from ".";
-import GalioTheme, { withGalio } from "./theme";
-
-class Checkbox extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      checked: props.initialValue
-    };
-
-    this.renderChecked = this.renderChecked.bind(this);
-    this.spaceAround = this.spaceAround.bind(this);
-  }
-
+import { Icon, Text } from 'galio-framework';
+import GalioTheme, { withGalio } from './theme';
+
+function Checkbox(props) {
+  const [checked, setchecked] = React.useState(props.initialValue);
+  React.useEffect(() => {
+    props.onChange(checked);
```

Comment:
the useEffect() function is getting triggered at every re-render of the component and when the component is first rendered? I'm not sure if triggering the onChange when the component first mounts is a good idea? What do you think?

Reply:
Please reply.